### PR TITLE
Make latest botocore tests green

### PR DESCRIPTION
**Pull request by potiuk**
The latest botocore tests are conflicting with a few requirements and until apache-beam upcoming version is released we need to do some manual exclusions. Those exclusions should make latest botocore test green again.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow-dev [hunk 1]
```diff
@@ -0,0 +1,196 @@
+<!DOCTYPE html>
```
Comment (Lee-W) [inline airflow-dev]: May I know what is this file for?

**Review by Lee-W — COMMENTED** [review]
is this for opensearch? or opensearch is part of apache-beam

File: airflow-dev (hunk 1 above)
Comment (potiuk) [reply airflow-dev]: Ah... some left-over accidentally committed. Will remove.

**Comment by potiuk** [comment]
> is this for opensearch? or opensearch is part of apache-beam

This is because currently the tests for "latest" botocore have conflicting dependencies and we are workarounding it - by temporarily disabling dependencies that have those conflicts.

In this case the problem is `apache.beam` that (currently released version)  has reqeusts < 3 but != 2.32.* (due to some problem with their test harness - apparently requests did not work with the docker test harness they used). The problem with that is that growing number of our dependencies (for example opensearch) has `requests >= 2.32` and when we attempt to upgrade botocore - through a ripple-effect of dependencies pulling other dependencies, this leads to unsolvable conflicts (because latest botocore will not work with earlier versions of those dependencies that can work with requests 2.31.*:

```  
  Uninstalled 3 packages in 320ms
   - aiobotocore==2.13.2
   - s3fs==2024.6.1
   - yandexcloud==0.313.0
  + uv pip install --python /usr/local/bin/python --upgrade boto3 botocore 'oss2>=2.14.0' 'cryptography<43.0.0' 'requests!=2.32.*,<3.0.0,>=2.24.0'
  Resolved 19 packages in 32.61s
  Prepared 3 packages in 325ms
  Uninstalled 3 packages in 913ms
  Installed 3 packages in 140ms
   - boto3==1.34.131
   + boto3==1.35.2
   - botocore==1.34.131
   + botocore==1.35.2
   - requests==2.32.3
   + requests==2.31.0
  + set +x

  opensearch-py 2.7.0 has requirement requests<3.0.0,>=2.32.0, but you have requests 2.31.0.
```


The apache-beam problem has been solved and merged as I opened https://github.com/apache/beam/issues/32080 - 2 weeks ago so that they remove the limitation - and they merged it literally yesterday - https://github.com/apache/beam/pull/32236 

So by removing opensearch from latest botocore tests, we termporarily make it works now - but once apache-beam 2.59.0 is released, we should be able to bring opensearch back to those tests.

**Comment by Lee-W** [comment]
got it! It makes sense now. Thanks for the details!

File: airflow-dev (hunk 1 above)
Comment (potiuk) [reply airflow-dev]: removed

**Comment by potiuk** [comment]
>  got it! It makes sense now. Thanks for the details!

Yes. Dependencies are hard. Airlfow dependencies doubly so.

**Comment by potiuk** [comment]
uv.lock also commited - by mistake now (removed it)

**Review by Lee-W — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Review by utkarsharma2 — APPROVED** [review]

**Pull request merged**